docs/19_security_audit_document.md

```diff
@@ -0,0 +1,275 @@
+
+Each document contains data associated with a specific user profile.
+
+#### Document structure
+
+| Field           | Description                                  |
+|-----------------|----------------------------------------------|
+| `selectedTheme` | A theme selected by the specific user.       |
+
+### Security rules and tests
+
+The application uses the `Firebase Cloud Firestore Security Rules` to protect the database. The `Security Rules` describe certain conditions needed to access or modify the database. The `Firebase Security Rules` provide the main protection for the data used in the project.
+
+Consider the following subsections that describe the security rules for the `Firebase Cloud Firestore` collections:
+
+#### [`project_groups`](#the-project_groups-collection) collection security rules:
+
+| Operation         | Security Rules                         |
+|-------------------|----------------------------------------|
+| `read`, `delete`  | `Authorization`                        | 
+| `create`, `update`| `Authorization`, `isProjectGroupValid` | 
+
+
+#### [`projects`](#the-projects-collection) collection security rules:
+
+| Operation          | Security Rules                    |
+|--------------------|-----------------------------------|
+| `read`             | `Authorization`                   |
+| `create`, `update` | `Authorization`, `isProjectValid` |
+| `delete`           | `Prohibited`                      |
+
+#### [`build`](#the-build-collection) collection security rules:
+
+| Operation          | Security Rules                  |
+|--------------------|---------------------------------|
+| `read`             | `Authorization`                 |
+| `create`, `update` | `Authorization`, `isBuildValid` |
+| `delete`           | `Prohibited`                    |
+
+#### [`user_profiles`](#the-user_profiles-collection) collection security rules:
+
+| Operation | Security Rules                                           |
+|-----------|----------------------------------------------------------|
+| `get`     | `Authorization`, `isDocumentOwner`                       |
+| `write`   | `Authorization`, `isDocumentOwner`, `isUserProfileValid` |
+| `list`    | `Prohibited`                                             |
+| `delete`  | `Prohibited`                                             |
+
+#### [`feature_config`](#the-feature_config-collection) collection security rules:
+
+| Operation | Security Rules |
+|-----------|----------------|
+| `read`    | `Allowed`      |
+| `write`   | `Prohibited`   |
+
+#### [`allowed_email_domains`](#the-allowed_email_domains-collection) collection security rules:
+
+| Operation       | Security Rules |
+|-----------------|----------------|
+| `read`, `write` | `Prohibited`   |
+
+These rules are covered with tests which are located in the [`metrics/firebase/test/firestore/security_rules`](https://github.com/platform-platform/monorepo/tree/master/metrics/firebase/test/firestore/rules) folder.
```

Comment:
We can keep it in the `Security Rules Testing` subsection below the "collections".